Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -508,10 +510,84 @@ extension BrowserViewController: TopToolbarDelegate {
         self.present(container, animated: true)
       }
     }
+    
+    shields.showSubmitReportView = { [weak self] shieldsViewController in
+      shieldsViewController.dismiss(animated: true) {
+        guard let url = shieldsViewController.tab.url else { return }
+        self?.showSubmitReportView(for: url, isPrivateMode: shields.tab.isPrivate)
+      }
+    }
+    
     let container = PopoverNavigationController(rootViewController: shields)
     let popover = PopoverController(contentController: container, contentSizeBehavior: .preferredContentSize)
     popover.present(from: topToolbar.locationView.shieldsButton, on: self)
   }
+  
+  private func showSubmitReportView(for url: URL, isPrivateMode: Bool) {
```

Comment:
You should be able to just get the current private mode from the `privateBrowsingManager` in BVC, no need to pass it in

Reply:
Done